osbs/api.py

```diff
@@ -189,19 +188,42 @@ def _panic_msg_for_more_running_builds(self, build_config_name, builds):
             (build_config_name, builds)
         return msg
 
-    def _create_build_config_and_build(self, build_request):
-        # TODO: test this method more thoroughly
-        build_json = build_request.render()
-        api_version = build_json['apiVersion']
-        if api_version != self.os_conf.get_openshift_api_version():
-            raise OsbsValidationException("BuildConfig template has incorrect apiVersion (%s)" %
-                                          api_version)
+    def _get_existing_build_config(self, build_config):
+        """
+        Uses the given build config to find an existing matching build config.
+        Build configs are a match if:
+        - metadata.name are equal
+        OR
+        - metadata.labels.git-repo-name AND metadata.labels.git-branch are equal
+        """
 
-        build_config_name = build_json['metadata']['name']
+        git_labels = (
+            ('git-repo-name', build_config['metadata']['labels']['git-repo-name']),
+            ('git-branch', build_config['metadata']['labels']['git-branch']),
+        )
+        name = build_config['metadata']['name']
+
+        queries = (
+            (self.os.get_build_config_by_labels, git_labels),
+            (self.os.get_build_config, name),
+        )
+
+        existing_bc = None
+        for func, arg in queries:
+            try:
+                existing_bc = func(arg)
+                # build config found
```

Comment:
What happens if there is a build config matching the name we created, but it has conflicting git-repo-name and/or git-branch labels? i.e. if we created a name for a build config that already exists